### ISSUE TITLE
`looksLikeProse` ANDs independent vetoes, so no signal can outvote another

### ISSUE THREAD

**Issue by Radiergummi** [Radiergummi/rfc-reader#44]
Blocked on the instrumentation, ground-truth and boundary-labelling work. Filing now so the
reasoning is recorded; **do not start this until those have produced evidence that it is
needed.**

### The actual problem with the current design

Not that the thresholds are badly chosen — that the signals are combined wrongly.
`looksLikeProse` (`LegacyTextParser.swift:497`) ANDs a set of independent absolute vetoes:

```swift
guard indent <= 6, (0...8).contains(firstLineIndent) else { return false }
…
if content.contains(artworkPattern) { return false }
```

A block indented 7 is artwork no matter how perfectly it reads as sentences. A paragraph
containing `->` or `---` is artwork on that basis alone, against every other signal agreeing.

These signals genuinely interact, but nothing in the code lets one outvote another, and there is
no way to express that as thresholds without inventing the weights by hand — which is precisely
the thing worth automating.

### Shape of the work

The features already exist or nearly exist in the parser: `indent`, `firstLineIndent`,
right-margin alignment, `hasColumnGutter`, `internalGapCount`, alphanumeric-to-box-drawing
ratio, line-final punctuation, `readsLikeSentences`. Perhaps fifteen in total. Refactoring
`looksLikeProse` into a feature vector plus a decision function is a small change.

Train off-device — Create ML's tabular classifiers, or scikit-learn/LightGBM, whichever is less
friction. **Ship the weights, not a model.** A gradient-boosted tree is nested `if`s; logistic
regression is a dot product.

Linux support is not the reason (that constraint has been dropped — a hosted macOS runner is
fine, and training locally is fine). The reasons are:

- **Determinism.** `corpus/report.json` diffing only works if the classifier is reproducible.
  A framework model that Apple updates underneath us turns every diff into noise and destroys
  the one tool we have for judging whether a change was a net win.
- **One code path.** `corpus-build` and the app share the parser; a plain function needs no
  protocol seam, no injected classifier, no second implementation to keep in sync.
- **Explainability.** When a block is wrong today you can point at `guard indent <= 6` and
  argue about it. Feature attributions on a small tree preserve most of that; an opaque
  framework call does not.

### Required if this ships

Pin the model version in `report.json`. Without it, a retrain silently invalidates every prior
corpus comparison.
